src/Microsoft.DotNet.Build.Tasks/Targets/tests.targets

```diff
@@ -49,12 +52,39 @@
     <StartArguments Condition="'$(StartArguments)'==''">$(VSStartArguments)</StartArguments>
     <DebugEngines>{2E36F1D4-B23C-435D-AB41-18E608940038}</DebugEngines>
   </PropertyGroup>
+  
+  <ItemGroup>
+    <TestTargetFramework Include="ASP.NetCore, version=v5.0">
+      <Folder>aspnetcore50</Folder>
+    </TestTargetFramework>
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackagesConfigs Include="$(ProjectPackagesConfigFile)" />
+    <PackagesConfigs Include="$(TestRuntimePackageConfig)" />
+  </ItemGroup>
+
+  <Target Name="DiscoverTestInputs">
+    <ItemGroup>
+      <RunTestsForProjectInputs Include="@(PackagesConfigs)" />
+      <RunTestsForProjectInputs Include="@(ReferenceCopyLocalPaths)" />
+      <RunTestsForProjectInputs Include="@(Content)" />
+      <RunTestsForProjectInputs Include="@(IntermediateAssembly)" />
+      <RunTestsForProjectInputs Include="@(_DebugSymbolsIntermediatePath)" />
+      <RunTestsForProjectInputs Include="@(AllItemsFullPathWithTargetPath)" />
+    </ItemGroup>
+  </Target>
 
   <!-- On command line run unit tests on CoreCLR after the Test target -->
   <Target Name="RunTestsForProject"
+          DependsOnTargets="DiscoverTestInputs"
```

Comment:
Ah, I see. If I remove DependsOnTargets="DiscoverTestInputs", then RunTestsForProject is skipped because it has no inputs. If I put it back, the target is executed as expected on the first run, and not executed on the second run, as expected. This tells me that the inputs for RunTestsForProject are not being evaluated until after DiscoverTestInputs has executed; otherwise, executing DiscoverTestInputs would have no effect on whether RunTestsForProject is executed.